src/backend/src/controllers/users.controllers.ts

```diff
@@ -1,182 +1,95 @@
-        emailId,
-        userSettings: { create: {} }
-      },
-      ...authUserQueryArgs
-    });
-    user = createdUser;
-  }
+  static async getSingleUser(req: Request, res: Response, next: NextFunction) {
+    try {
+      const userId: number = parseInt(req.params.userId);
+      const requestedUser = await UsersService.getSingleUser(userId);
 
-  // register a login
-  await prisma.session.create({
-    data: {
-      userId: user.userId,
-      deviceInfo: req.headers['user-agent']
+      res.status(200).json(requestedUser);
+    } catch (error: unknown) {
+      next(error);
     }
-  });
-
-  const token = generateAccessToken({ userId: user.userId, firstName: user.firstName, lastName: user.lastName });
-  res.cookie('token', token, { httpOnly: true, sameSite: 'none', secure: true });
-
-  return res.status(200).json(authenticatedUserTransformer(user));
-};
-
-// for dev login only!
-export const logUserInDev = async (req: any, res: any) => {
-  if (process.env.NODE_ENV === 'production') return res.status(400).json({ message: 'Cant dev login on production!' });
-  if (!req.body || !req.body.userId) return res.status(400).json({ message: 'Invalid Body' });
-
-  const { body } = req;
-  const { userId } = body;
+  }
 
-  const user = await prisma.user.findUnique({
-    where: { userId },
-    ...authUserQueryArgs
-  });
+  static async getUserSettings(req: Request, res: Response, next: NextFunction) {
+    try {
+      const userId: number = parseInt(req.params.userId);
+      const settings = await UsersService.getUserSettings(userId);
 
-  if (!user) {
-    return res.status(400).json({ message: 'That user does not exist' });
+      res.status(200).json(settings);
+    } catch (error: unknown) {
+      next(error);
+    }
   }
 
-  // register a login
-  await prisma.session.create({
-    data: {
-      userId: user.userId,
-      deviceInfo: req.headers['user-agent']
-    }
-  });
+  static async updateUserSettings(req: Request, res: Response, next: NextFunction) {
+    try {
+      const { defaultTheme, slackId } = req.body;
+      const userId = parseInt(req.params.userId);
 
-  return res.status(200).json(authenticatedUserTransformer(user));
-};
+      const settings = await UsersService.updateUserSettings(userId, defaultTheme, slackId);
 
-export const updateUserRole = async (req: Request, res: Response) => {
-  const errors = validationResult(req);
-  if (!errors.isEmpty()) {
-    return res.status(400).json({ errors: errors.array() });
+      res.status(200).json({ message: `Successfully updated settings for user ${userId}.` });
+    } catch (error: unknown) {
+      next(error);
+    }
   }
 
-  const targetUserId: number = parseInt(req.params.userId);
+  static async logUserIn(req: Request, res: Response, next: NextFunction) {
+    try {
+      const idToken = req.body.id_token;
+      const header = req.headers['user-agent'];
 
-  const { body } = req;
+      const user = await UsersService.logUserIn(idToken, header!);
```

Comment:
oh this is kinda weird cuz you need the user and token for the response. teh controller should have no logic though, so generate the token in the service and then return { user, token } from the service so you can use both here in the response